kedro-org/kedro-viz
# Retrieve viz url

## Description
Fixes #1851 
Design is [here]( https://www.figma.com/file/3kSpvIO1veLKfy9qHxuXwF/Kedro-WIP?type=design&node-id=1411-87299&mode=design&t=bSwcXxbq95GPPfud-4)
![retrieve-url](https://github.com/kedro-org/kedro-viz/assets/32060364/086160e8-183d-4ccb-9103-0f73be427079)





## Development notes

- we store store the fields including hosting platform, bucket name and endpoint URL in [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) once the user publishes it. The format is as follow
```
{
   aws: {
       bucket_name: ...,
       platform: 'aws',
       endpoint: '....',
   },
   gcp: {....},
   azure: {....}
}
```

- Render the published content view will depends on whether there's only 1 hosting platform stored in localStorage or more than 1
- Current logic is summarised as the chart below

<img width="554" alt="Screenshot 2024-05-21 at 16 48 53" src="https://github.com/kedro-org/kedro-viz/assets/32060364/56b6cfae-2d10-4251-9b72-4711cd9c65b6">

## QA notes
You can test the UI using Gitpod. I've implemented the logic prior to the check that validates whether the submitted platform is valid or not. As a result, you can use any fabricated names and URLs, the system will display an error message.



## Checklist
